From 5cd05bdcc19c0675d6875fecc0896c56cee610ad Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@freescale.com>
Date: Fri, 21 Dec 2012 13:31:29 -0600
Subject: [PATCH 6/8] ENGR00238058: Vold: Ignore the mmcblkXboot before
 updating mDiskNumParts.

Ignore mmcblkXboot before updating mDiskNumParts
as NPARTS is 0 for mmcblkXboot disk.

Signed-off-by: Nitin Garg <nitin.garg@freescale.com>
---
 DirectVolume.cpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/DirectVolume.cpp b/DirectVolume.cpp
index 27168c1..d5badd7 100644
--- a/DirectVolume.cpp
+++ b/DirectVolume.cpp
@@ -158,6 +158,12 @@ void DirectVolume::handleDiskAdded(const char *devpath, NetlinkEvent *evt) {
 
     const char *tmp = evt->findParam("NPARTS");
     const char *name = evt->findParam("DEVNAME");
+
+    if (name && (!strncmp(name, "mmcblk0boot", 11) || (!strncmp(name, "mmcblk1boot", 11)))) {
+        SLOGD("Ignore the eMMC boot disk:%s", name);
+        return;
+    }
+
     if (tmp) {
         mDiskNumParts = atoi(tmp);
     } else {
@@ -165,11 +171,6 @@ void DirectVolume::handleDiskAdded(const char *devpath, NetlinkEvent *evt) {
         mDiskNumParts = 1;
     }
 
-    if (name && !strncmp(name, "mmcblk0boot", 11)) {
-        SLOGD("Ignore the eMMC boot disk:%s", name);
-        return;
-    }
-
     if (mDiskNumParts > MAX_PARTITIONS)
         mDiskNumParts = MAX_PARTITIONS;
 
-- 
1.8.0

